packages/aws-cdk-lib/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -1840,6 +1880,45 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
 
     return errors;
   }
+
+  private renderInstanceMaintenancePolicy(
+    maxHealthPercentage?: number,
+    minHealthPercentage?: number,
+  ): CfnAutoScalingGroup.InstanceMaintenancePolicyProperty | undefined {
+    if (maxHealthPercentage === undefined && minHealthPercentage === undefined) {
+      return;
+    }
+
+    if (
+      (maxHealthPercentage !== undefined && minHealthPercentage === undefined)
+      || (maxHealthPercentage === undefined && minHealthPercentage !== undefined)
+    ) {
+      throw new Error(`Both or neither of instanceMaintenancePolicyMinHealthPercentage and instanceMaintenancePolicyMaxHealthPercentage must be specified, got instanceMaintenancePolicyMinHealthPercentage: ${minHealthPercentage} and instanceMaintenancePolicyMaxHealthPercentage: ${maxHealthPercentage}`);
+    }
+    if ((maxHealthPercentage !== -1 && minHealthPercentage === -1) || (maxHealthPercentage === -1 && minHealthPercentage !== -1)) {
+      throw new Error(`Both instanceMaintenancePolicyMinHealthPercentage and instanceMaintenancePolicyMaxHealthPercentage must be -1 to clear the previously set value, got instanceMaintenancePolicyMinHealthPercentage: ${minHealthPercentage} and instanceMaintenancePolicyMaxHealthPercentage: ${maxHealthPercentage}`);
+    }
+    if (maxHealthPercentage !== undefined && maxHealthPercentage !== -1 && (maxHealthPercentage < 100 || maxHealthPercentage > 200)) {
+      throw new Error(`instanceMaintenancePolicyMaxHealthPercentage must be between 100 and 200, or -1 to clear the previously set value, got ${maxHealthPercentage}`);
+    }
+    if (minHealthPercentage !== undefined && minHealthPercentage !== -1 && (minHealthPercentage < 0 || minHealthPercentage > 100)) {
+      throw new Error(`instanceMaintenancePolicyMinHealthPercentage must be between 0 and 100, or -1 to clear the previously set value, got ${minHealthPercentage}`);
+    }
+    if (
+      maxHealthPercentage !== undefined
+      && minHealthPercentage !== undefined
+      && maxHealthPercentage !== -1
```

Comment:
```suggestion
      maxHealthPercentage !== -1
```
The case where both values are `undefined` is already covered by validation.

Reply:
No, logically, it is, but type inference in TypeScript does not work and a Lint error is printed, saying that the possibility of undefined is left at `maxHealthPercentage - minHealthPercentage > 100`.

The following code would work, but I avoided it because of the nesting of `if`s.

```ts
    if (maxHealthPercentage === undefined && minHealthPercentage === undefined) {
      return;
    } else if (maxHealthPercentage !== undefined && minHealthPercentage !== undefined) {
      if ((maxHealthPercentage !== -1 && minHealthPercentage === -1) || (maxHealthPercentage === -1 && minHealthPercentage !== -1)) {
        throw new Error(`Both maintenancePolicyMinHealthPercentage and maintenancePolicyMaxHealthPercentage must be -1 to clear the previously set value, got maintenancePolicyMinHealthPercentage: ${minHealthPercentage} and maintenancePolicyMaxHealthPercentage: ${maxHealthPercentage}`);
      }
      if (maxHealthPercentage !== -1 && (maxHealthPercentage < 100 || maxHealthPercentage > 200)) {
        throw new Error(`maintenancePolicyMaxHealthPercentage must be between 100 and 200, or -1 to clear the previously set value, got ${maxHealthPercentage}`);
      }
      if (minHealthPercentage !== -1 && (minHealthPercentage < 0 || minHealthPercentage > 100)) {
        throw new Error(`maintenancePolicyMinHealthPercentage must be between 0 and 100, or -1 to clear the previously set value, got ${minHealthPercentage}`);
      }
      if (
        maxHealthPercentage !== -1
        && minHealthPercentage !== -1
        && maxHealthPercentage - minHealthPercentage > 100
      ) {
        throw new Error(`The difference between maintenancePolicyMinHealthPercentage and maintenancePolicyMaxHealthPercentage cannot be greater than 100, got ${maxHealthPercentage - minHealthPercentage}`);
      }
    } else {
      throw new Error(`Both or neither of maintenancePolicyMinHealthPercentage and maintenancePolicyMaxHealthPercentage must be specified, got maintenancePolicyMinHealthPercentage: ${minHealthPercentage} and maintenancePolicyMaxHealthPercentage: ${maxHealthPercentage}`);
    }
```